naucse_render/course.py

```diff
@@ -0,0 +1,193 @@
+"""
+Retreive course meta-information
+
+Reads source YAML files and merges them to one JSON, with
+render info for items.
+"""
+
+from pathlib import Path
+import datetime
+import textwrap
+
+from .load import read_yaml
+from .markdown import convert_markdown
+
+
+API_VERSION = 1
+
+
+def encode_dates(value):
+    if isinstance(value, datetime.date):
+        return value.isoformat()
+    elif isinstance(value, dict):
+        return {k: encode_dates(v) for k, v in value.items()}
+    elif isinstance(value, (list, tuple)):
+        return [encode_dates(v) for v in value]
+    elif isinstance(value, (str, int, bool, type(None))):
+        return value
+    raise TypeError(value)
+
+
+# XXX: Clarify the version
```

Comment:
what version? it says `API_VERSION = 1` above.

Reply:
Ah, right. That's a remnant of when I tried to add versioning as well.
The PR is big enough without that, so I'll leave it to the future.